### PR TITLE
WEB-1507 Update Nginx & Clean up Project

### PR DESCRIPTION
**Pull request by pzelnip**
Addresses the fact we were running an old, vulnerable version of Nginx in the Docker image.

Took the opportunity to substantially clean up the project, the requirements.txt file was installing a *ton* of libraries that were not used, the Makefile still had old Beanstalk build stuff, added Safety to the Jenkins build, etc.

Note that this requires that https://github.com/bambora/terraform-bamborana-cde/pull/299 be merged & applied.

### CODE REVIEW

File: JenkinsFile [hunk 1]
```diff
@@ -52,6 +48,24 @@ timestamps {
                     }
                 }
 
+                stage('safetycheck') {
+                    sh """echo 'foobar'"""
```
Comment (jstep) [inline JenkinsFile]: Debugging statement?

Comment (pzelnip) [reply JenkinsFile]: Nice catch! Will remove.

File: server/Dockerfile [hunk 2]
```diff
@@ -17,13 +8,10 @@ RUN apk add --no-cache --update python3 \
   && rm -rf /root/.cache \
   && apk add --no-cache --update \
      build-base python3-dev libffi-dev postgresql-dev \
-     nginx supervisor \
+     nginx supervisor logrotate \
```
Comment (dan-sparkes-bambora) [inline server/Dockerfile]: I see we are installing logrotate, I'm not seeing any configuration for it.  We should either configure it or not install it

File: server/Makefile [hunk 3]
```diff
@@ -1,67 +1,27 @@
...
-
-	aws s3 cp $(ZIP) s3://$(EB_BUCKET)/$(ZIP)
-
-	# Create new app version with zipped Dockerrun file
-	aws elasticbeanstalk create-application-version --application-name merchant-api-demo \
-	--version-label $(VERSION) \
-	--source-bundle S3Bucket=$(EB_BUCKET),S3Key=$(ZIP) \
-	--region us-west-2
+shell: build
+	docker run --rm -it $(IMAGE_NAME):$(VERSION) /bin/ash
```
Comment (dan-sparkes-bambora) [inline server/Makefile]: /bin/ash ?

Comment (pzelnip) [reply server/Makefile]: It's what was there before.  Prefer sh?

**Pull request merged**